Debian8 package test failure

Example failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.3+packaging-tests/58/console

This is a strange failure, from what I can see in the logs, the packaging test for debian 8 retry continually to hit ES, but keep timing out.

```
10:09:53 .Jul 17, 2019 3:07:55 PM org.elasticsearch.packaging.test.DefaultNoJdkDebBasicTests setup
10:09:53 INFO: [test80DeletePID_DIRandRestart]: before test
10:09:53 Jul 17, 2019 3:07:56 PM org.apache.http.impl.execchain.RetryExec execute
10:09:53 INFO: I/O exception (org.apache.http.NoHttpResponseException) caught when processing request to {}->http://localhost:9200: The target server failed to respond
10:09:53 Jul 17, 2019 3:07:56 PM org.apache.http.impl.execchain.RetryExec execute
10:09:53 INFO: Retrying request to {}->http://localhost:9200
10:09:53 .Jul 17, 2019 3:08:04 PM org.elasticsearch.packaging.test.DefaultNoJdkDebBasicTests setup
10:09:53 INFO: [test81CustomPathConfAndJvmOptions]: before test
10:09:53 Jul 17, 2019 3:08:04 PM org.apache.http.impl.execchain.RetryExec execute
10:09:53 INFO: I/O exception (org.apache.http.NoHttpResponseException) caught when processing request to {}->http://localhost:9200: The target server failed to respond
10:09:53 Jul 17, 2019 3:08:04 PM org.apache.http.impl.execchain.RetryExec execute
10:09:53 INFO: Retrying request to {}->http://localhost:9200
```

This goes on for a while and then the build fails with the following:
```
10:09:53 There was 1 failure:
10:09:53 1) test93ElasticsearchNodeCustomDataPathAndNotEsHomeWorkDir(org.elasticsearch.packaging.test.DefaultNoJdkLinuxTarTests)
10:09:53 java.lang.RuntimeException: Command was not successful: [bash -c echo y | /tmp/elasticsearch/bin/elasticsearch-node unsafe-bootstrap] result: <org.elasticsearch.packaging.util.Shell$Result exitCode = [1] stdout = [------------------------------------------------------------------------
10:09:53 
10:09:53     WARNING: Elasticsearch MUST be stopped before running this tool.
10:09:53 
10:09:53 ] stderr = [future versions of Elasticsearch will require Java 11; your Java version from [/usr/lib/jvm/java-8-openjdk-amd64/jre] does not meet this requirement
10:09:53 Exception in thread "main" ElasticsearchException[failed to lock node's directory, is Elasticsearch still running?]; nested: LockObtainFailedException[Lock held by another program: /tmp/nodes/0/node.lock];
10:09:53 	at org.elasticsearch.cluster.coordination.ElasticsearchNodeCommand.processNodePathsWithLock(ElasticsearchNodeCommand.java:85)
10:09:53 	at org.elasticsearch.cluster.coordination.ElasticsearchNodeCommand.execute(ElasticsearchNodeCommand.java:121)
10:09:53 	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
10:09:53 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
10:09:53 	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:77)
10:09:53 	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
10:09:53 	at org.elasticsearch.cli.Command.main(Command.java:90)
10:09:53 	at org.elasticsearch.cluster.coordination.NodeToolCli.main(NodeToolCli.java:47)
10:09:53 Caused by: org.apache.lucene.store.LockObtainFailedException: Lock held by another program: /tmp/nodes/0/node.lock
10:09:53 	at org.apache.lucene.store.NativeFSLockFactory.obtainFSLock(NativeFSLockFactory.java:130)
10:09:53 	at org.apache.lucene.store.FSLockFactory.obtainLock(FSLockFactory.java:41)
10:09:53 	at org.apache.lucene.store.BaseDirectory.obtainLock(BaseDirectory.java:45)
10:09:53 	at org.elasticsearch.env.NodeEnvironment$NodeLock.<init>(NodeEnvironment.java:214)
10:09:53 	at org.elasticsearch.cluster.coordination.ElasticsearchNodeCommand.processNodePathsWithLock(ElasticsearchNodeCommand.java:77)
10:09:53 	... 7 more
10:09:53 ]>
10:09:53 	at __randomizedtesting.SeedInfo.seed([62E284B94E79E33D:6C1F8ACAA8D82CE9]:0)
10:09:53 	at org.elasticsearch.packaging.util.Shell.runScript(Shell.java:94)
10:09:53 	at org.elasticsearch.packaging.util.Shell.run(Shell.java:61)
10:09:53 	at org.elasticsearch.packaging.test.ArchiveTestCase.test93ElasticsearchNodeCustomDataPathAndNotEsHomeWorkDir(ArchiveTestCase.java:438)
10:09:53 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
10:09:53 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
10:09:53 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
10:09:53 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
10:09:53 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
10:09:53 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
10:09:53 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
10:09:53 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
10:09:53 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
10:09:53 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
10:09:53 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
10:09:53 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
10:09:53 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$2.evaluate(ThreadLeakControl.java:404)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSuite(RandomizedRunner.java:708)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner.access$200(RandomizedRunner.java:138)
10:09:53 	at com.carrotsearch.randomizedtesting.RandomizedRunner$2.run(RandomizedRunner.java:629)
10:09:53 
10:09:53 FAILURES!!!
10:09:53 Tests run: 278,  Failures: 1
```

--- a/qa/os/src/test/java/org/elasticsearch/packaging/util/Archives.java
+++ b/qa/os/src/test/java/org/elasticsearch/packaging/util/Archives.java
@@ -314,8 +314,8 @@ public class Archives {
         assertThat(pid, not(isEmptyOrNullString()));
 
         final Shell sh = new Shell();
-        Platforms.onLinux(() -> sh.run("kill -SIGTERM " + pid));
-        Platforms.onWindows(() -> sh.run("Get-Process -Id " + pid + " | Stop-Process -Force"));
+        Platforms.onLinux(() -> sh.run("kill -SIGTERM " + pid + "; tail --pid=" + pid + " -f /dev/null"));
+        Platforms.onWindows(() -> sh.run("Get-Process -Id " + pid + " | Stop-Process -Force; Wait-Process -Id " + pid));
     }
 
 }

